homeassistant/components/derivative/config_flow.py

```diff
@@ -31,49 +30,54 @@
 )
 
 UNIT_PREFIXES = [
-    {"value": "none", "label": "none"},
-    {"value": "n", "label": "n (nano)"},
-    {"value": "µ", "label": "µ (micro)"},
-    {"value": "m", "label": "m (milli)"},
-    {"value": "k", "label": "k (kilo)"},
-    {"value": "M", "label": "M (mega)"},
-    {"value": "G", "label": "T (tera)"},
-    {"value": "T", "label": "P (peta)"},
+    selector.SelectOptionDict(value="none", label="none"),
+    selector.SelectOptionDict(value="n", label="n (nano)"),
+    selector.SelectOptionDict(value="µ", label="µ (micro)"),
+    selector.SelectOptionDict(value="m", label="m (milli)"),
+    selector.SelectOptionDict(value="k", label="k (kilo)"),
+    selector.SelectOptionDict(value="M", label="M (mega)"),
+    selector.SelectOptionDict(value="G", label="G (giga)"),
+    selector.SelectOptionDict(value="T", label="T (tera)"),
+    selector.SelectOptionDict(value="P", label="P (peta)"),
 ]
 TIME_UNITS = [
-    {"value": TIME_SECONDS, "label": "Seconds"},
-    {"value": TIME_MINUTES, "label": "Minutes"},
-    {"value": TIME_HOURS, "label": "Hours"},
-    {"value": TIME_DAYS, "label": "Days"},
+    selector.SelectOptionDict(value=TIME_SECONDS, label="Seconds"),
+    selector.SelectOptionDict(value=TIME_MINUTES, label="Minutes"),
+    selector.SelectOptionDict(value=TIME_HOURS, label="Hours"),
+    selector.SelectOptionDict(value=TIME_DAYS, label="Days"),
 ]
 
 OPTIONS_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_ROUND_DIGITS, default=2): selector.selector(
-            {
-                "number": {
-                    "min": 0,
-                    "max": 6,
-                    "mode": "box",
-                    CONF_UNIT_OF_MEASUREMENT: "decimals",
-                }
-            }
+            selector.SelectorType.NUMBER,
+            selector.NumberSelectorDict(
```

Comment:
Ok, so this is more nitpicking, so feel free to decline/ignore.

```py
            selector.NumberSelectorDict(
```

All these types, passing the selector configuration are called `XSelectorDict`. Which is fine, as the type is a typed dictionary, however, I think the `Dict` postfix is not needed.

However, we do need a postfix to avoid name collisions. So how about just `Config`?

- `NumberSelectorDict` -> `NumberSelectorConfig`
- `SelectSelectorDict` -> `SelectSelectorConfig`
- etc

It makes its purpose a bit more clear?

Reply:
I like it!

Reply:
and nitpick away, if this doesn't make things easier, or if there are ways to make things easier, I am all for it